pkg/registry/pod/etcd/etcd.go

```diff
@@ -60,7 +62,7 @@ type REST struct {
 }
 
 // NewStorage returns a RESTStorage object that will work against pods.
-func NewStorage(s storage.Interface, useCacher bool, k client.ConnectionInfoGetter, proxyTransport http.RoundTripper) PodStorage {
+func NewStorage(s storage.Interface, nodeStore *nodeetcd.REST, useCacher bool, k client.ConnectionInfoGetter, proxyTransport http.RoundTripper) PodStorage {
```

Comment:
Please define and take an interface for this, rather than making a hard dependency on the node store.
e.g. something like:

```go
type HostLocater interface {
  HostURL(pod *api.Pod) (*url.URL, error)
}
```